File: leetcode/4.寻找两个有序数组的中位数.js
/*
 * @lc app=leetcode.cn id=4 lang=javascript
 *
 * [4] 寻找两个有序数组的中位数
 */
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number}
 */
var findMedianSortedArrays = function(nums1, nums2) {
    let arr;
    if (!nums1 || !nums1.length) {
        arr = nums2;
    } else if (!nums2 || !nums2.length) {
        arr = nums1;
    } else {
        arr = nums1.concat(nums2).sort((a, b) => b - a);
    }

    if (arr.length % 2 == 0) {
        return (arr[arr.length / 2] + arr[arr.length / 2 - 1]) / 2;
    } else {
        return arr[(arr.length - 1) / 2];
    }
};

File: leetcode/4.寻找两个有序数组的中位数.py
#
# @lc app=leetcode.cn id=4 lang=python3
#
# [4] 寻找两个有序数组的中位数
#


class Solution:
    def findMedianSortedArrays(self, nums1: List[int], nums2: List[int]) -> float:
        res = nums1 + nums2
        res.sort()
        n = len(res) // 2
        return res[n] if len(res) % 2 == 1 else (res[n]+res[n-1])/2
